server/src/main/java/org/opensearch/common/xcontent/XContentHelper.java

```diff
@@ -68,8 +68,13 @@ public static XContentParser createParser(NamedXContentRegistry xContentRegistry
             if (compressedInput.markSupported() == false) {
                 compressedInput = new BufferedInputStream(compressedInput);
             }
-            final XContentType contentType = XContentFactory.xContentType(compressedInput);
-            return XContentFactory.xContent(contentType).createParser(xContentRegistry, deprecationHandler, compressedInput);
+            try {
+                final XContentType contentType = XContentFactory.xContentType(compressedInput);
+                return XContentFactory.xContent(contentType).createParser(xContentRegistry, deprecationHandler, compressedInput);
+            } catch (Exception e) {
+                compressedInput.close();
+                throw e;
+            }
```

Comment:
```suggestion
            } finally {
                compressedInput.close();
            }
```

Reply:
The `InputStream` looks like need to exist after the method returns, otherwise there will be test failure:
```
    AzureBlobStoreRepositoryTests. testIndicesDeletedFromRepository
    AzureBlobStoreRepositoryTests. testMultipleSnapshotAndRollback
    AzureBlobStoreRepositoryTests. testRequestStats
    AzureBlobStoreRepositoryTests. testSnapshotWithLargeSegmentFiles
```